tests/System.Text.Primitives.Tests/ParserTests.cs

```diff
@@ -4,472 +4,983 @@
 
 namespace System.Text.Primitives.Tests
 {
-    public class ParserTests
+	public class ParserTests
     {
-        private byte[] UtfEncode(string s)
+		private byte[] UtfEncode(string s)
```

Comment:
I think you should just either have both `Uft8Encode` and a `Uft16Encode` methods, or just a single `UtfEncode` method with a bool switch. I'd prefer the former (two explicitly-named methods).